arm-iothub/2017-01-19/swagger/iothub.json

```diff
@@ -964,9 +964,452 @@
+          {
+            "$ref": "#/parameters/api-version"
+          },
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "name": "resourceGroupName",
+            "description": "The name of the resource group that contains the IoT hub.",
+            "in": "path",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "resourceName",
+            "in": "path",
+            "description": "The name of the IoT hub.",
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The body contains all the certificate list.",
+            "schema": {
+              "$ref": "#/definitions/CertificateListDescription"
+            }
+          },
+          "default": {
+            "description": "DefaultErrorResponse",
+            "schema": {
+              "$ref": "#/definitions/ErrorDetails"
+            }
+          }
+        },
+        "deprecated": false
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Devices/IotHubs/{resourceName}/certificates/{certificateName}": {
+      "get": {
+        "tags": [ "GET" ],
+        "summary": "Get the certificate.",
+        "description": "Returns the certificate.",
+        "operationId": "Certificate_Get",
+        "parameters": [
+          {
+            "$ref": "#/parameters/api-version"
+          },
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "name": "resourceGroupName",
+            "description": "The name of the resource group that contains the IoT hub.",
+            "in": "path",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "resourceName",
+            "in": "path",
+            "description": "The name of the IoT hub.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "certificateName",
+            "in":"path",
+            "description":"The name of the certificate",
+            "required":true,
+            "type":"string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The body contains the certificate.",
+            "schema": {
+              "$ref": "#/definitions/CertificateDescription"
+            }
+          },
+          "default": {
+            "description": "DefaultErrorResponse",
+            "schema": {
+              "$ref": "#/definitions/ErrorDetails"
+            }
+          }
+        },
+        "deprecated": false
+      },
+      "put": {
+        "tags": [ "PUT" ],
+        "summary": "Upload the certificate to the IoT hub.",
+        "description": "Adds new or replaces existing certificate.",
+        "operationId": "Certificate_CreateOrUpdate",
```

Comment:
I think it is. If the operation can return 201, most likely, it's a [long running operation](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/creating-swagger.md#longrunning). We don't care much how long the operation will run but we do care what the API will be generated from the swagger. if it's a long running operation, usually, we will generate a function which can handle 201. Otherwise, a user of the API should write his own pooling mechanism for the function.